engine/class_modules/sc_death_knight.cpp

```diff
@@ -6439,6 +6483,16 @@ struct swarming_mist_buff_t : public buff_t
 
     add_invalidate( CACHE_DODGE );
   }
+  void expire_override( int expiration_stacks, timespan_t remaining_duration ) override
+  {
+    buff_t::expire_override( expiration_stacks, remaining_duration );
+    death_knight_t* p = debug_cast< death_knight_t* >( player );
+    if ( p -> legendary.insatiable_hunger.ok() )
+    {
+      insatiable_damage -> execute();
+      insatiable_hunger_rp_spent = 0;
+    }
```

Comment:
I'd reset on cast start rather than cast ends, because I'm not sure whether expire_override is called if a buff is cancelled for the end of a simulation (so you may have to reset it there too, which resetting on cast circumvents)